packages/core/getUserAvailability.ts

```diff
@@ -122,8 +135,39 @@ export const getCurrentSeats = (eventTypeId: number, dateFrom: Dayjs, dateTo: Da
 
 export type CurrentSeats = Awaited<ReturnType<typeof getCurrentSeats>>;
 
+export const getUserAvailability = async (
+  query: {
+    withSource?: boolean;
+    username?: string;
+    userId?: number;
+    dateFrom: string;
+    dateTo: string;
+    eventTypeId?: number;
+    afterEventBuffer?: number;
+    beforeEventBuffer?: number;
+    duration?: number;
+  },
+  initialData?: {
+    user?: User;
+    eventType?: EventType;
+    currentSeats?: CurrentSeats;
+    rescheduleUid?: string | null;
+    currentBookings?: (Pick<Booking, "id" | "uid" | "userId" | "startTime" | "endTime" | "title"> & {
+      eventType: Pick<
+        PrismaEventType,
+        "id" | "beforeEventBuffer" | "afterEventBuffer" | "seatsPerTimeSlot"
+      > | null;
+      _count?: {
+        seatsReferences: number;
+      };
+    })[];
+  }
+) => {
```

Comment:
when writing the wrapper functions, no need to respecify all the parameters, you can use typescript `Parameters<typeof _getUserAvailability>` and the same approach used in sentry wrapper with `...args`, example:

```ts
const getEventType = async (...args: Parameters<typeof _getEventType>) => {
  return monitorCallback(_getEventType, args);
};
```

Reply:
In the newest commit. Thanks for the suggestion!